### PR TITLE
Task/DES-1435 - Allow inter-sorting of different collection types

### PR DESCRIPTION
**Pull request by Strmiska**
Allow collections to be sortable among other collection types
Update UI to display sorted collections
Fixes to various bugs in the Data Diagram and UI

**TO BE MERGED WITH Task/DES-1358**

### CODE REVIEW

File: designsafe/static/scripts/ng-designsafe/services/project-service.js [hunk 1]
```diff
@@ -162,7 +162,7 @@ export function ProjectService(httpi, $interpolate, $q, $state, $uibModal, Loggi
         } else if (project.value.projectType === 'field_recon') {
             requiredSets = [
                 'planning_set',
-                'socialscience_set',
+                // 'socialscience_set',
```
Comment (fnets) [inline designsafe/static/scripts/ng-designsafe/services/project-service.js]: This line can probably be deleted, too.

File: designsafe/static/scripts/projects/components/project-tree/project-tree.component.js [hunk 2]
```diff
@@ -466,22 +500,26 @@ class ProjectTreeCtrl {
      * @method
      *
      * Build tress for every Mission in this class' `this.project`.
-     * The hierarchy is build like so:
-     *
-     * + Mission 1
+     * The hierarchy is built like so:
```
Comment (fnets) [inline designsafe/static/scripts/projects/components/project-tree/project-tree.component.js]: I thought these comments were really helpful!

File: designsafe/static/scripts/ng-designsafe/services/project-service.js (hunk 1 above)
Comment (Strmiska) [reply designsafe/static/scripts/ng-designsafe/services/project-service.js]: Ah this is okay, this whole section is being overwritten in another PR.

File: designsafe/static/scripts/projects/components/project-tree/project-tree.component.js (hunk 2 above)
Comment (Strmiska) [reply designsafe/static/scripts/projects/components/project-tree/project-tree.component.js]: Thanks :D

File: designsafe/static/scripts/data-depot/components/projects/pipeline-selection/pipeline-selection.component.js [hunk 3]
```diff
@@ -38,7 +38,7 @@ class PipelineSelectionCtrl {
             this.ProjectEntitiesService.listEntities({ uuid: this.projectId, name: 'all' })
         ]).then(([project, listing, entities]) => {
             this.browser.project = project;
-            this.prepProject();
+            // this.prepProject();
```
[the remark this answers is not in the document]
Comment (Strmiska) [reply designsafe/static/scripts/data-depot/components/projects/pipeline-selection/pipeline-selection.component.js]: Removing!

**Pull request merged**